src/main/java/ulcrs/controllers/SessionController.java

```diff
@@ -14,36 +25,82 @@
 
 public class SessionController extends BaseController {
 
+    private static final String WORKSPACE_PATH = "workspace/";
+
     @Override
     public RouteGroup routes() {
         return () -> {
             before("/*", (request, response) -> log.info("endpoint: " + request.pathInfo()));
-            get("/", this::getSessionList, gson::toJson);
+            get("/", this::getSessionList, sessions -> {
+                Gson gson = new GsonBuilder().excludeFieldsWithoutExposeAnnotation().setPrettyPrinting().create();
+                return gson.toJson(sessions);
+            });
             get("/:name", this::getSession, gson::toJson);
-            post("/:name", this::updateSession, gson::toJson);
+            post("/:name", this::saveSession, gson::toJson);
             delete("/:name", this::deleteSession, gson::toJson);
         };
     }
 
     private List<String> getSessionList(Request request, Response response) {
-        // TODO: return names of sessions with limit specified
-        return null;
+        response.type(CONTENT_TYPE_JSON);
+        int limit = Integer.valueOf(request.params("limit"));
```

Comment:
Not sure how Spark handles parameters and query parameters, but is this a query parameter or a body param? I.e. is this `/ulcrs/session/?limit=<limit>`, or is it sent in the body of the HTTP GET request? I would think we'd want the params to be query params, not from the request body.

Reply:
Changed to request.queryParams("limit")